src/main/java/backend/resource/TurboLabel.java

```diff
@@ -197,18 +289,51 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         TurboLabel that = (TurboLabel) o;
-        return actualName.equals(that.actualName) && colour.equals(that.colour);
+        return fullName.equals(that.fullName) && colour.equals(that.colour);
     }
 
     @Override
     public int hashCode() {
-        int result = actualName.hashCode();
+        int result = fullName.hashCode();
         result = 31 * result + colour.hashCode();
         return result;
     }
 
     @Override
     public int compareTo(TurboLabel o) {
-        return actualName.compareTo(o.getActualName());
+        return fullName.compareTo(o.getFullName());
+    }
+
+    private static String joinWith(String group, String name, boolean exclusive) {
+        return group + (exclusive ? EXCLUSIVE_DELIMITER : NONEXCLUSIVE_DELIMITER) + name;
+    }
+
+    private Grouping initGroupMembership() {
+        if (getDelimiter(fullName).isPresent()) {
+            return getDelimiter(fullName).get().equals(EXCLUSIVE_DELIMITER) 
+                ? Grouping.EXCLUSIVE : Grouping.NON_EXCLUSIVE;
+        }
+        return Grouping.NONE;
     }
+
+    /**
+     * Determines and returns the group that labelName belongs to
+     * A labelName is considered to be in a group if getDelimiter(labelName).isPresent() is true.
+     * @return
+     */
+    private String initGroupName() {
+        if (!isInGroup()) return "";
+        return fullName.substring(0, fullName.indexOf(getDelimiter(fullName).get()));
+    }
```

Comment:
This is better (stateless, and the method name fits the functionality): `String extractGroupName(String fullName)`